src/main/java/com/google/sps/data/UserQuery.java

```diff
@@ -0,0 +1,60 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import com.google.sps.util.SearchUtils;
+import java.util.ArrayList;
+
+/**
+ * Represents the separated user query data which can be used to run different Datastore
+ * query requests.
+ */
+public class UserQuery {
+  private ArrayList<String> tagsAndIngredients;
+  private ArrayList<String> authors;
```

Comment:
Does it ever actually make sense to have multiple authors in the query? From what I understand from what you've discussed (correct me if I'm wrong), these attributes will be combined with the AND operator, ie if the user searches for `"#sweet #chocolate @olivia"` then you'll return recipes that have both tag=sweet **and** tag=chocolate **and** author=olivia. Since each recipe can only have one author, a query with multiple authors will always return empty results, right?

Reply:
I don't think it does, and I did this more thinking about implementing author search to be explicitly an **OR** operator in the future. So the back end could handle these differently, and add some more meaningful search options for a user.

Reply:
Okay, that sounds fine. If you are using AND for some elements of the search and OR for other parts, just be sure that's obvious from the frontend. Basically, the user shouldn't be surprised by how the search works.